### PR TITLE
Add smoke test

### PR DESCRIPTION
**Pull request by ejguan**
Per title.
Smoke test will be used by project Nova to test compiled TorchData

### CODE REVIEW

File: test/smoke_test/smoke_test.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import argparse
+
+import torchdata
+import torchdata.dataloader2
+import torchdata.datapipes
```
Comment (dahsh) [inline test/smoke_test/smoke_test.py:12]: Do we need `datapipes` here?

Comment (ejguan) [reply test/smoke_test/smoke_test.py:12]: Sorry, I am not sure what do you mean

**Review by NivekT — APPROVED** [review]
LGTM!

Question:
This looks a bit different. from the one in TorchVision (not saying this is wrong), there is probably an internal post/guide that I missed? How does this work?

**Comment by DanilBaibak** [comment]
@NivekT we use these smoke tests to test binaries:

1. First build binaries - https://github.com/pytorch/test-infra/blob/main/.github/workflows/build_wheels_macos.yml#L118
2. Install the current build - https://github.com/pytorch/test-infra/blob/main/.github/workflows/build_wheels_macos.yml#L164
3. Than try to import the main library blocks - https://github.com/pytorch/test-infra/blob/main/.github/workflows/build_wheels_macos.yml#L177

Smoke tests may different for different domain libraries. Any ideas are welcome 😄

**Review by DanilBaibak — APPROVED** [review]
LGTM

**Pull request closed without merging**